falcon/response.py

```diff
@@ -425,6 +432,9 @@ def set_cookie(self, name, value, expires=None, max_age=None,
         if http_only:
             self._cookies[name]['httponly'] = http_only
 
+        if same_site:
+            self._cookies[name]['samesite'] = same_site
```

Comment:
I'm admittedly a bit divided on the validation question. @kgriffs  thoughts?
I would say we do actually validate some parameters, for instance, `max_age` is cast to `int` etc. Now that we are on Python 3, defining an `enum` with supported values is another option to consider. OTOH we are not using `enum`s anywhere else in Falcon (at least not yet).
NB if we decide to validate anything, we could also consider supporting the experimental `'None'` value too.

Reply:
The easiest way to check if a value is correct.
e.g
```
if same_site and same_site in {'Lax', 'Strict', 'None'}
```
otherwise, skip it, that's it;)

Reply:
`...same_site.lower() in {`lax', 'strict', 'none`}` - the value is case insensitive, right?

Reply:
if the value will be case insensitive, we will need to do something like your wrote.

It will allow us, to set value in different ways.

e.g
```
resp.set_cookie('samesite', 'Lax')
# OR
resp.set_cookie('samesite', 'lax')
```

I like it.

Reply:
I don't think validation will hurt from a performance standpoint and is a simple thing to do. An enum may provide some benefits from a type-checking perspective if/when we add type hints to the framework, but on the other hand using an enum would require few more keystrokes from the developer vs. a simple string, and then we would have to spend a few extra cycles serializing out the enum when setting the cookie attribute.

Regardless, I think we really should do validation because a typo in this case can impact security (considering that the user agent maps unknown values to the "None" behavior).

My preference would be to keep the `str`-based approach (don't use an enum) and do the validation as proposed above, e.g.:

```py
same_site.lower() in {'lax', 'strict', 'none'}
```

Reply:
I like the enum idea, since it obviates the need to do any value checking. With respect to time spent serializing the enum to string (which is effectively just a dict lookup under the hood), that would be offset at least somewhat by the avoidance of explicit string value checking.

Reply:
Also, if we're really trying to squeeze every last cycle out of the code, then we might find that the lookup is faster in a tuple than in a set, with such a small N. Worth measuring?
```
same_site.lower() in ('lax', 'strict', 'none')
```

Reply:
I, unfortunately, disagree here, I don't think so that we need `enum` module.

This is only one place where we do this kind of check if I remember correctly.

And the main reason - some people will be wondering why we are using this kind of enum, but this is just a check.

IMO.

Reply:
@RonRothman Lookup is fastest on `frozenset`s and `set`s if you cache them in a global variable. Benchmarked that before.

Reply:
@vytas7 Ack, (frozen)set it is.